### PR TITLE
Core/Movement: Fix movement handler crash during ReadMovementInfo check

### PR DESCRIPTION
**Pull request by chaodhib**
**Changes proposed:**

-  Fix a rare edge case that can lead to a null pointer crash

**Target branch(es):** 3.3.5/master

- [X] 3.3.5
- [ ] master

**Issues addressed:**

Should close https://github.com/TrinityCore/TrinityCore/issues/26642

**Tests performed:**

Tested IG.

**Known issues and TODO list:** (add/remove lines as needed)

- [ ] 
- [ ]

### CODE REVIEW

File: src/server/game/Server/WorldSession.cpp [hunk 1]
```diff
@@ -926,7 +927,7 @@ void WorldSession::ReadMovementInfo(WorldPacket &data, MovementInfo* mi)
             mi->RemoveMovementFlag((maskToRemove));
     #endif
 
-    Unit* mover = GetGameClient()->GetActivelyMovedUnit();
+    Unit* mover = GetPlayer()->GetGUID() == mi->guid ? GetPlayer() : ObjectAccessor::GetUnit(*GetPlayer(), mi->guid);
```
Comment (jackpoz) [inline src/server/game/Server/WorldSession.cpp:930]: could you please add an if check on mover after this line and handle that case too ? whether it's return + log error or ASSERT (even if a return + log is always better if possible so it doesn't crash the whole server)

Comment (Shauren) [reply src/server/game/Server/WorldSession.cpp:930]: mi->guid is client controlled, it must not be ASSERT

Comment (chaodhib) [reply src/server/game/Server/WorldSession.cpp:930]: @jackpoz will do tonight.

@Shauren 

> how does this change affect cheating prevention? 

WorldSession::ReadMovementInfo is called in a movement handler only after all the permission checks are made

Comment (Shauren) [reply src/server/game/Server/WorldSession.cpp:930]: Wouldnt this mean that there is a data consistency error in GameClient state? `_allowedMovers` has a guid but `_activelyMovedUnit` is not set?

Comment (chaodhib) [reply src/server/game/Server/WorldSession.cpp:930]: @jackpoz done. i choose the assert approach and not just returning because otherwise weird things will happen after the function will return.

@Shauren It might. I'll look into it. It could also be explained by the following edge case:
on login, there is a window of time (after the server initialize the player and sends `CreateObject2` to the client and before the clients sends `CMSG_SET_ACTIVE_MOVER`) during which the GameClient's `_allowedMovers` contains the player but `_activelyMovedUnit` is null. If the client sends an ACK message during that time window, we get the situation you described

Comment (chaodhib) [reply src/server/game/Server/WorldSession.cpp:930]: In the meantime, let's merge this already?

**Comment by jackpoz** [comment]
@chaodhib does https://github.com/TrinityCore/TrinityCore/issues/26642#issuecomment-872005053 help ?

File: src/server/game/Server/WorldSession.cpp (hunk 1 above)
Comment (jackpoz) [reply src/server/game/Server/WorldSession.cpp:930]: Did you check the crashlogs mentioned below ?

**Comment by jackpoz** [comment]
it seems @offl and @Aokromes have a lot of faith in this PR, let's merge it then and see what happens next
